src/libraries/System.Private.CoreLib/src/Resources/Strings.resx

```diff
@@ -3410,6 +3410,9 @@
   <data name="SemaphoreSlim_Wait_TimeoutWrong" xml:space="preserve">
     <value>The timeout must represent a value between -1 and Int32.MaxValue, inclusive.</value>
   </data>
+  <data name="SemaphoreSlim_Wait_TimeSpanTimeoutWrong" xml:space="preserve">
+    <value>The timeout must represent a value between -1 and the maximum allowed timer duration.</value>
```

Comment:
Can the error message specify the maximum allowed duration?

Also, these APIs do not create timers so the error message should not mention timer.